src/esp/gfx/CubeMap.h

```diff
@@ -36,7 +36,12 @@ class CubeMap {
     Depth = 1,
     // TODO: ObjectId
     // TODO: HDR color
+
+    // you will have to update it when new type is added
+    Count = 2,
```

Comment:
```suggestion
    Count
```

Keep it without a number, that's the trick. The compiler will automatically assign a value larger than the previous enum value and thus you don't have to explicitly take care of this.

Reply:
@mosra : Honestly, I knew this trick. And what I did before was 1) I would not use the number, 2) I would just use the `enum` instead of the `enum class` so that you do `texture_[TextureType::Color]` directly without cast.

Later, Facebook coding convention does not allow me to do it. All the items within the enum has to have a number.  And later, in the habitat project, `enum` is not allowed, and I can only use `enum class`.

Reply:
> Facebook coding convention requires me to put a number to each item in the enum

It's actively harmful in this case and results in extra maintenance burden, so I'd suggest to not follow that rule here, no matter how they insist on that. `enum class` is fine tho.